java/vector/src/test/java/org/apache/arrow/vector/ValueVectorDataPopulator.java

```diff
@@ -0,0 +1,379 @@
+/*
```

Comment:
can you demonstrate the usage of this class in at least one instance in a different package tests?  I want to make sure it is reusable.

Reply:
Now this class was located in vector test package, should we move it into vector main package?

Reply:
I think th easiest thing to do would be to move it to main, under a new java package "org.apache.arrow.vector.testing".  Another alternative (which I don't think would work) is create an entirely new mvn submodule and put it in main there.

I think @jacques-n might have concerns around this, so lets wait for him to chime in (I think I posted earlier about a possible way of exposing the testing jar to other packages but it seemed hacky).